Update TypeSpec Authoring section in scenario 2 spec (#13991)
Azure/azure-sdk-tools: tools/azsdk-cli/docs/specs/0-scenario-2.spec.md

   - Optionally remediate missing or out-of-date tools with `--auto-install` option
   - **Note**: This stage carries over from Scenario 1 and will need to be revisited to ensure it works correctly for [Brand New Packages](#brand-new-package)

2. **TypeSpec Authoring** → `azsdk_typespec_generate_authoring_plan`

>    - AI-powered assistance for authoring or modifying TypeSpec API specifications

Reviewer: @copilot , according to the typespec authoring design spec, https://github.com/Azure/azure-sdk-tools/blob/main/tools/azsdk-cli/docs/specs/typespec-authoring.spec.md, it is a skill called: azure-typespec-author. not an mcp tool, and it is agent mode only.

> 2. **TypeSpec Authoring** → TypeSpec Authoring Skills **(Agent Mode only)**

Reviewer: ```suggestion
2. **TypeSpec Authoring** → `azsdk_typespec_authoring` skill**(Agent Mode only)**
```

   - Leverages Azure SDK knowledge base for guidelines-compliant code
   - Helps with ARM resources, versioning, routing, and compliance fixes

>    - **Note**: Skills are transparently invoked by GitHub Copilot based on TypeSpec context detection

Reviewer: ```suggestion
   - **Note**: `azsdk_typespec_authoring` skill is transparently invoked by GitHub Copilot based on TypeSpec context detection
```

Before or during SDK generation, developers may need to author or modify TypeSpec API specifications. Scenario 2 introduces AI-powered assistance for TypeSpec authoring that leverages the Azure SDK knowledge base to generate standards-compliant code.

> **Skill:** TypeSpec Authoring Skills (leveraging `azsdk_typespec_generate_authoring_plan` MCP tool)

Reviewer: ```suggestion
**Skill:** `azsdk_typespec_authoring` skill (leveraging `azsdk_typespec_generate_authoring_plan` MCP tool)
```

Use --auto-install flag to remediate issues automatically.
```

### 2. TypeSpec Generate Authoring Plan

> **Note:** This CLI command is used internally by the TypeSpec Authoring Skills in GitHub Copilot and is not intended for direct user invocation in Scenario 2. Users should interact with GitHub Copilot in Agent Mode, which will automatically invoke this skill when working with TypeSpec files.

Reviewer: @ronniegeraghty, may I learn more details of your concern? I might not get it currently. Currently `azsdk_typespec_authoring` is a skill which underline leverages `azsdk_typespec_generate_authoring_plan` MCP tool. 
 From my understanding (@chunyu3 can correct me), there is no configuration difference between `azsdk_typespec_generate_authoring_plan` MCP tool and other MCP tool, once the MCP server is started, `azsdk_typespec_generate_authoring_plan` MCP tool is enabled.
